File: app/src/main/java/com/example/tiangou/a1218test/MainActivity.java
package com.example.tiangou.a1218test;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {


    ScratchView scratchView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);



        scratchView = (ScratchView) findViewById(R.id.scratch_view);

        scratchView.setmEraseStatusListener(new ScratchView.EraseStatusListener() {
            @Override
            public void onProgress(int percent) {
                android.util.Log.d("ScratchView", "onProgress  " + percent);
            }

            @Override
            public void onCompleted(View view) {
                android.util.Log.d("ScratchView", "onCompleted");

            }
        });
    }
}

File: kotlincoroutinedemo/src/main/java/com/tiangou/kotlincoroutinedemo/MainActivity.kt
package com.tiangou.kotlincoroutinedemo

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import butterknife.ButterKnife
import butterknife.OnClick

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ButterKnife.bind(this)
    }


    @OnClick(R.id.sequential_coroutine_jobs)
    fun startSequentialCoroutineJobsAct() {

        val intent = Intent(this, SequentialCoroutineJobsAct::class.java)
        startActivity(intent)

    }

    @OnClick(R.id.deferred_async_await_successive)
    fun startAct1() {

        val intent = Intent(this, DeferredAsyncAwaitSuccessiveAct::class.java)
        startActivity(intent)

    }



}
